front/app/modules/commercial/customizable_homepage_banner/admin/CallToActionButtonSettings.tsx

```diff
@@ -14,39 +14,62 @@ import {
 } from 'services/appConfiguration';
 
 interface SigninStateProps {
-  options: string[];
+  options: (CTASignedOutButton | CTASignedInButton)[];
   currentValue: string;
   signInState: 'not_signed_in' | 'signed_in';
+  labelMessage: MessageDescriptor;
 }
 
 const CallToActionSettingsSigninState = ({
```

Comment:
I would rename this component. It's hard to tell from the component name what it does. And it should also be extracted in a separate file ideally.

Reply:
> And it should also be extracted in a separate file ideally

Sure, will split all files by components a bit later when it's clear how the state can be managed.

> I would rename this component.

I agree. I don't like the word `state` because it's confusingly similar to react state. What do you think about `CTASettingsBySigninStatus`?

Reply:
Or you put the SectionField and Label stuff in the signed in/signed out components (sometimes a little repetition is ok, especially because it's not going to be more than signed in and out. If it is, we can still optimize later if it's needed).

Then the reused part is just the radio buttons and you can call it just that SettingRadios or SettingRadioButtons. Another benefit: we have two fewer props here

Reply:
For me, both variants (my initial one and the one you propose) are both fine with their own advantages and disadvantages. And maybe this discussion and this refactoring is a bit (just a bit :smile: ) waste of time, but sure, I will change it :slightly_smiling_face:

Reply:
Haha, didn't need to refactor, but it helped with choosing a name 😄